src/book/01-team/05-robocup/03-debriefs.mdx

```diff
@@ -6,9 +6,171 @@ description: Debriefs and takeaways from our performances at RoboCup.
+
+- Having a way to rapidly prototype ideas in NUClear (i.e. python modules)
+- A “digital whiteboard”
+  - Keep track of what needs to be done, and who’s doing what
+  - Keep track of specific dev things which don’t fit in NUbook - e.g. when is the next game, everyone needs to `./b target generic` again because something broke, etc.
+  - Develop a system to assign tasks to people GitHub Issues/Projects? Trello?
+- Get newer people confident before RoboCup because when we’re needing to be speedy there isn’t a lot of time to think about helping shy/less confident members.
+- Improved walk engine
+- Vision: how useful is improving the vision data collection tool? If we use Webots again, maybe? Not useful for real world. Ideas for real world are
+  - Multimodal CycleGAN
+  - Motion capture data
+- All new vision members need to be able to train, run, test visual mesh (with a dummy dataset)
+- Consider more post-vision processing including work on green horizon throw outs.
+- Optimise odometry values, maybe for when motion capture is set up
+- Develop head behaviour to include searching for goals as well as the ball
+- Proper state machine for behaviour to handle all cases - Kick-off in Normal, Ball drop, Other things. Infrastructure to begin with rather than actual implementation of what to do in that stage.
+- Implement isolated tests in Webots for modules.
+- More NUbook pages. Documentation is important for new members to learn things quickly
+- We need the Director, a system that will improve our robots decision making when it comes to what behaviours run at a given time
+- Have a look at implementing ROS bridge, a system with a huge number of libraries that we can use if we can interface with it
+- Investigate PlotJuggler to be used as potential alternative to NUsight charts
+- Numerical optimization for Inverse Kinematics
+- Highly accurate system identification so our model best translates from Webots to the real world
+- Path planning/tracking (vector fields that include obstacles or something similar)
+- Develop a simple program that uses sliders for tuning YAML files for our scripts
+- Multi-state walk engine, which starts in a slower state where it is more stable, then once it has started moving gradually increment the speed
+- Behaviour visualisation. Either some indication in existing NUsight views of the state of the robot, or a new NUsight behaviour view, with some indication of graphs of previous states and possible future states.
+- Potentially a different NUsight view for different providers e.g. a NUsight view for the walk engine behaviour provider
+- Change `GameController.cpp` to use `Single` instead of `Sync` - `Single` can drop messages, but `Sync` will not
+- Fix `BallDetector` to properly let the system know if it can detect no balls
+- Set up a different config for our playing role, so we debug and test on one role, and play on the role for playing - to prevent breaking config changes like the network traffic filter
```

Comment:
'debug and test on one set of configs and play on another set of configs' sounds more like what this is about?